AnimationSystem/AnimationSystem.cc

```diff
@@ -0,0 +1,1098 @@
+  void initAnimationSystem(float *scratchStack) { // only need init once globally
+    if (!isInitedAnimationSystem) {
+      // note: Need be the same order with `constants.h/enum SingleAnimationIndex`.
+      singleAnimations.push_back(animationAll["idle.fbx"]); // Idle
+      singleAnimations.push_back(animationAll["Crouch Idle.fbx"]); // CrouchIdle
+      singleAnimations.push_back(animationAll["jump.fbx"]); // Jump
+      singleAnimations.push_back(animationAll["jump_double.fbx"]); // DoubleJump
+      singleAnimations.push_back(animationAll["falling.fbx"]); // FallLoop
+      singleAnimations.push_back(animationAll["treading water.fbx"]); // Float
+      animationGroups.push_back(singleAnimations);
+
+      // ---
+
+      // 8 directions walk animations ---
+      // note: Need be the same order with `constants.h/enum WalkAnimationIndex`.
+      walkAnimations.push_back(animationAll["walking.fbx"]); // Forward
+      walkAnimations.push_back(animationAll["walking backwards.fbx"]); // Backward
+      walkAnimations.push_back(animationAll["left strafe walking.fbx"]); // Left
+      walkAnimations.push_back(animationAll["right strafe walking reverse.fbx"]); // LeftMirror
+      walkAnimations.push_back(animationAll["right strafe walking.fbx"]); // Right
+      walkAnimations.push_back(animationAll["left strafe walking reverse.fbx"]); // RightMirror
+      animationGroups.push_back(walkAnimations);
+
+      // 8 directions run animations ---
+      // note: Need be the same order with `constants.h/enum RunAnimationIndex`.
+      runAnimations.push_back(animationAll["Fast Run.fbx"]); // Forward
+      runAnimations.push_back(animationAll["running backwards.fbx"]); // Backward
+      runAnimations.push_back(animationAll["left strafe.fbx"]); // Left
+      runAnimations.push_back(animationAll["right strafe reverse.fbx"]); // LeftMirror
+      runAnimations.push_back(animationAll["right strafe.fbx"]); // Right
+      runAnimations.push_back(animationAll["left strafe reverse.fbx"]); // RightMirror
+      animationGroups.push_back(runAnimations);
+
+      // 8 directions crouch animations ---
+      // note: Need be the same order with `constants.h/enum CrouchAnimationIndex`.
+      crouchAnimations.push_back(animationAll["Sneaking Forward.fbx"]); // Forward
+      crouchAnimations.push_back(animationAll["Sneaking Forward reverse.fbx"]); // Backward
+      crouchAnimations.push_back(animationAll["Crouched Sneaking Left.fbx"]); // Left
+      crouchAnimations.push_back(animationAll["Crouched Sneaking Right reverse.fbx"]); // LeftMirror
+      crouchAnimations.push_back(animationAll["Crouched Sneaking Right.fbx"]); // Right
+      crouchAnimations.push_back(animationAll["Crouched Sneaking Left reverse.fbx"]); // RightMirror
+      animationGroups.push_back(crouchAnimations);
+
+      // -------------------------------------------------------------------------
+
+      // note: Need be the same order with `constants.h/enum ActivateAnimationIndex`.
```

Comment:
This note is good, but...

Is it possible to iterate the enum determinisitcally so we don't have to maintain 2 places in code?

Reply:
Used `initializer list`s instead of `enum`s, to put animation groups' declarations in one place, all in `constants.h` now.

https://github.com/webaverse/app-wasm/compare/8a301a809cf32605c96cf72b6d4c587143e1e4eb...ab2b22d3e2c43874a4f6645648d9e36b1b316d1a